internals/cli/cmd_add_test.go

```diff
@@ -100,6 +100,9 @@ services:
 			} else if path == triggerLayerPath {
 				c.Assert(err, check.ErrorMatches, "triggered")
 			} else if path == unreadableLayerPath {
+				if os.Getuid() == 0 {
+					c.Skip("requires running as non-root users")
```

Comment:
It's a bit weird to skip the test only once you get to this code path. We could skip the whole test, but I'd probably just only do this `IsPermission` assert if it's non-root. For example:

```go
if os.Getuid() != 0 {
    c.Assert(os.IsPermission(err), check.Equals, true)
}
```